bacon/model/StorageCouchBaseDB.swift

```diff
@@ -0,0 +1,136 @@
+                throw error
+            } else {
+                throw InitializationError(message: "unable to access embedded \(name.rawValue) mobile database.")
+            }
+        }
+    }
+
+    private static func getOrCreateFolderPath(for url: URL) throws -> String {
+        let folderPath = url.path
+        let fileManager = FileManager.default
+        if !fileManager.fileExists(atPath: folderPath) {
+            // Create folder if non-existent
+            do {
+                try fileManager.createDirectory(atPath: folderPath,
+                                                withIntermediateDirectories: true,
+                                                attributes: nil)
+            } catch {
+                throw InitializationError(message:
+                    "Database directory path is non-existent and encountered error creating directory: \(folderPath)")
+            }
+        }
+        return folderPath
+    }
+
+    private func createMutableDocument(from transaction: Transaction) throws -> MutableDocument {
+        do {
+            let transactionData = try transaction.asDictionary()
+            let transactionDocument = MutableDocument(data: transactionData)
+            return transactionDocument
+        } catch {
+            throw StorageError(message: "Transaction couldn't be encoded into MutableDocument.")
+        }
+    }
+
+    func saveTransaction(_ transaction: Transaction) throws {
+        do {
+            let transactionDocument = try createMutableDocument(from: transaction)
+            try transactionDatabase.saveDocument(transactionDocument)
+        } catch let error {
+            if error is StorageError {
+                throw error
+            } else {
+                throw StorageError(message: "Transaction couldn't be saved into database.")
+            }
+        }
+    }
+
+    func loadTransactions(ofType type: TransactionType, limit: Int) throws -> [Transaction] {
+        let query = QueryBuilder.select(SelectResult.all())
+                                .from(DataSource.database(transactionDatabase))
+                                .where(Expression.property("type").equalTo(Expression.string(type.rawValue)))
+                                .orderBy(Ordering.property("date").descending())
+                                .limit(Expression.int(limit))
+        do {
+            var transactions: [Transaction] = Array()
+            for result in try query.execute().allResults() {
+                guard let transactionDictonary = result.toDictionary()["transactions"]else {
+                    throw StorageError(message: "Transactions of type \(type) couldn't be loaded from database.")
+                }
+                let transactionData = try JSONSerialization.data(withJSONObject: transactionDictonary, options: [])
+                let dateFormatter = DateFormatter()
+                dateFormatter.locale = Locale(identifier: "en_US_POSIX")
```

Comment:
All these magic values can be refactored into static constants, or if it's project-wise, into constants.swift